mk/config.mk

```diff
@@ -29,10 +29,12 @@ TARGET_FLAGS += -DUSE_CONFIG_SOURCE
 endif
 
 CONFIG_REVISION := norevision
-ifeq ($(shell git -C $(CONFIG_DIR) diff --shortstat),)
-CONFIG_REVISION := $(shell git -C $(CONFIG_DIR) log -1 --format="%h")
+ifeq ($(shell git -C "$(CONFIG_DIR)" rev-parse --is-inside-work-tree 2>/dev/null),true)
+ifeq ($(strip $(shell git -C "$(CONFIG_DIR)" status --porcelain 2>/dev/null)),)
+CONFIG_REVISION := $(shell git -C "$(CONFIG_DIR)" rev-parse --short=12 HEAD 2>/dev/null)
```

Comment:
This will change CONFIG_REVISION format (from 7 to 12 chars).
config revision is mostly informative, %h is IMO more appropriate